### ISSUE TITLE
Add simple linear layer option as last layer of MLP

### ISSUE THREAD

**Issue by j-adamczyk** [pyg-team/pytorch_geometric#4552]
### 🚀 The feature, motivation and pitch

In the current `MLP` implementation all layers use the same activation, normalization and dropout. However, this means that for graph classification we actually can't use it.

This is because typically ReLU is used as an activation function. For the last layer, we actually want no activation (i.e. linear), so that we can use `cross_entropy` function, which applies softmax internally. No activation in the last layer therefore really applies softmax for loss calculation. We also can't use dropout, as we want all classes. Batch norm is also often omitted in this case.

Option for simple linear layer as the last layer would solve this problem. Implementation is simple:
- add `simple_linear_last_layer` argument or something similar
- set an attribute
- during forward phase do not apply activation, batch norm and dropout in the last layer if this attribute is true

Setting default value as `False` would trivially keep backwards compatibility.

### Alternatives

A more verbose alternative would be 3 options, to be able to turn off activation and/or batch norm and/or dropout, but such fine grained configuration is probably out of scope of this generic class.

**Comment by rusty1s** [comment]
This is indeed missing. Can you send a PR for this? This would be great.

Instead of `simple_linear_last_layer`, can we come about with a shorter name? :)

**Comment by j-adamczyk** [comment]
@rusty1s sure. Maybe `plain_last`?

**Comment by rusty1s** [comment]
That's better :)

**Comment by rusty1s** [comment]
Hey @j-adamczyk, just had a quick look once again and just noted that `MLP` utilizes `plain_last` by default, see https://github.com/pyg-team/pytorch_geometric/blob/master/torch_geometric/nn/models/mlp.py#L150. As such, we can either close this issue, or add this option nonetheless (with `plain_last=True` by default).

**Comment by j-adamczyk** [comment]
@rusty1s I looked at the code again and you're right, last layer is linear by default. However, I was quite surprised at this, as I absolutely assumed, based on typical MLP implementations, that all layers are identical. Just to clarify, two questions for the current implementation:

1. `MLP([32, 2])`. This means that we'll have a single linear layer, with no activation / dropout / batchnorm, just `nn.Linear(32, 16)`?
2. `MLP([32, 16, 2]`. This means that we'll have:
  - `nn.Linear(32, 16)`, with activation, dropout and batchnorm
  - `nn.Linear(16, 2)`, with no activation, dropout and batchnorm
    ?

**Comment by rusty1s** [comment]
Yes, that is correct.

**Comment by j-adamczyk** [comment]
This is quite surprising, based on my previous experiences with MLP implementations in CNN libraries. I would expect all layers to perform identically, unless specified otherwise. I see two solutions:
- add this to documentation, so that it's more explicit how MLP works
- add `plain_last` with default `True` value

Of course, we can do both. What do you think?

**Comment by rusty1s** [comment]
Yes, we should do both. Let me know if you want to integrate that. Otherwise, I can try to do it myself.

**Comment by j-adamczyk** [comment]
I will take care of that, this should be pretty small thing.
